nion/swift/test/DocumentModel_test.py

```diff
@@ -1977,7 +1977,7 @@ def test_undelete_connection(self):
             display_item = document_model.get_display_item_for_data_item(data_item)
             pick_data_item = document_model.get_pick_new(display_item)
             pick_display_item = document_model.get_display_item_for_data_item(pick_data_item)
-            self.assertEqual(1, len(document_model.connections))
+            self.assertEqual(2, len(document_model.connections))
```

Comment:
Why would there be two connections now instead of one?

Reply:
That's because this test is using `pick`, which has a second connection now. This is used to connect the sequence index of the spectrum image to the sequence index of the pick result. So a sequence of spectrum images will result in a sequence of pick data. The connection is always initialized, also for non-sequence spectrum images. See here for the code: https://github.com/Brow71189/nionswift/blob/073dae61216736879b72a21748bf46b6f578c3bd/nion/swift/model/DocumentModel.py#L2535

Reply:
I'm going to take a bit of time to think about this. I'm trying to eliminate connections overall - but I need to review my previous notes on this.

Reply:
Sure if there is a better replacement for connections I'm happy to use it. But I guess we need them in some way in order to do things like connecting a pick graphic to a display slice. Or are planning to replace this with a different system?

Reply:
I'm planning to deprecate all of the persistent Connections and replace with implicit connections that get built at load time. The computation has all of the required information to build the connection dynamically at load time. It's less general, but much lower maintenance. For now, this patch looks ok with two connections, though. I will update it when I make the implicit connection change.

Reply:
Note to future self: this will switch to implicit connections. Ok for now.